File: Python/program/dosen.py
# Saya Muhamad Firdaus [2101995] mengerjakan
# soal Latihan 4 dalam mata kuliah DPBO
# untuk keberkahanNya maka saya tidak melakukan kecurangan 
# seperti yang telah dispesifikasikan. Aamiin.

from sivitas import SivitasAkademik

class Dosen(SivitasAkademik):
    # Constructor
    def __init__(self, NIK, NIP, nama, jenis_kelamin, asal_universitas, email_edu, fakultas, pend_terakhir, keahlian):
        SivitasAkademik.__init__(self, NIK, nama, jenis_kelamin, asal_universitas, email_edu)
        self.NIP = NIP
        self.fakultas = fakultas
        self.pend_terakhir = pend_terakhir
        self.keahlian = keahlian
        
    # Setter dan Getter
    def setNIP(self, NIP):
        self.NIP = NIP
        
    def getNIP(self):
        return self.NIP
    
    def setFakultas(self, fakultas):
        self.fakultas = fakultas
                
    def getFakultas(self):
        return self.fakultas
    
    def setPendidikanTerakhir(self, pend_terakhir):
        self.pend_terakhir = pend_terakhir
                
    def getPendidikanTerakhir(self):
        return self.pend_terakhir
    
    def setKeahlian(self, keahlian):
        self.keahlian = keahlian
                
    def getKeahlian(self):
        return self.keahlian
                
    
                
    
                
    

    

File: cpp/program/dosen.cpp
// Saya Muhamad Firdaus [2101995] mengerjakan
// soal Latihan 4 dalam mata kuliah DPBO
// untuk keberkahanNya maka saya tidak melakukan kecurangan 
// seperti yang telah dispesifikasikan. Aamiin.

#include "header.hh"

// Constructor
Dosen::Dosen(){
    this->NIP = "";
    this->fakultas = "";
    this->pend_terakhir = "";
    this->keahlian = "";
}

// Constructor
Dosen::Dosen(string NIK, string NIP, string nama, string jenis_kelamin, string asal_universitas, string email_edu, string fakultas, string pend_terakhir, string keahlian)
    : SivitasAkademik(NIK, nama, jenis_kelamin, asal_universitas, email_edu){
    this->NIP = NIP;
    this->fakultas = fakultas;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
}

// Setter dan Getter
void Dosen::setNIP(string NIP){
    this->NIP = NIP;
}

string Dosen::getNIP(){
    return this->NIP;
}

void Dosen::setFakultas(string fakultas){
    this->fakultas = fakultas;
}
        
string Dosen::getFakultas(){
    return this->fakultas;
}
        
void Dosen::setPendidikanTerakhir(string pend_terakhir){
    this->pend_terakhir = pend_terakhir;
}
        
string Dosen::getPendidikanTerakhir(){
    return this->pend_terakhir;
}
        
void Dosen::setKeahlian(string keahlian){
    this->keahlian = keahlian;
}

string Dosen::getKeahlian(){
    return this->keahlian;
}

// Destructor
Dosen::~Dosen(){
    
}